deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml

```diff
@@ -34,15 +34,23 @@ spec:
             cpu:
               description: CPU defines set of CPU related parameters.
               properties:
+                enableIsolcpus:
```

Comment:
Do we really need the boolean? What about just checking the isolated string is nil or empty?

Reply:
the issue from what I could see is there's tuned partitioning that uses the Isolated CPUSet independently of whether or not we set the isolcpu kernel arg.

Reply:
@MarSik the Isolated cpu set is used in two different configs.

1. In a tuned config to ste teh isolated_cores cpu set.
```
[scheduler]
isolated_cores="4-7"
```

2. In a MachineConfig to set the IRQBALANCE_BANNED_CPUS=11110000 bitmask for irqbalance.

```
echo "IRQBALANCE_BANNED_CPUS=11110000" >> ${INITRD_GENERATION_DIR}/etc/sysconfig/irqbalance
```

Since item [2] here lets us execute arbitrary bash logic on the actually host, the inverse of `Reserved` is possible. However for item [1], since the tuned profile is generated at the cluster level, the only way to compute the inverse of `Reserved` is to know how many cpus are on the matching hosts, which isn't something we can get from the node objects themselves by default.

So, basically the Isolated option is required because there isn't a reasonable way (that i'm aware of) to calculate the inverse of `Reserved` at the cluster level. Because of this, we need the `EnableIsolcpu` boolean to indicate if that kernel arg should be set or not independently of the `Isolated` cpu set

make sense?

Reply:
@davidvossel is right regarding the difference of isolated_cores used in kargs and isolated_cores passed to the [tuned profie](https://github.com/openshift-kni/performance-addon-operators/blob/832b615823bd28807d0eef2748d808739f7b266e/build/assets/tuned/openshift-node-real-time-kernel#L22). 

> 2. In a MachineConfig to set the IRQBALANCE_BANNED_CPUS=11110000 bitmask for irqbalance.

the pre tuning script we have right now takes the reserved cpus list and inverts it to an isolated_cpumask.
On the other hand we [set](https://github.com/openshift-kni/performance-addon-operators/blob/832b615823bd28807d0eef2748d808739f7b266e/build/assets/scripts/pre-boot-tuning.sh#L37) CPUAffinity from the reserved cpu list.

Just as a reference, Tuned cpu-partitioning profile takes a list of isolated cpus and inverts them to non_isolated_cpus [list](https://github.com/redhat-performance/tuned/blob/bfe0eb91949d4e305de5c22bec913fc8318b3a33/profiles/cpu-partitioning/tuned.conf#L25) (or [mask](https://github.com/redhat-performance/tuned/blob/bfe0eb91949d4e305de5c22bec913fc8318b3a33/profiles/cpu-partitioning/tuned.conf#L28)) for setting CPUAffinity.  Supported py scripts for that can be found [here](https://github.com/redhat-performance/tuned/blob/master/tuned/profiles/functions/function_cpulist_invert.py).